stdlib/map.gr

```diff
@@ -274,8 +277,20 @@ export let fromList = (list) => {
   map
 }
 
+let setInArray = (array) => (i, key, value) => {
+  array[i] = (key, value)
+  i + 1
+}
+
+@disableGC
 export let toArray = (map) => {
-  Array.fromList(toList(map))
+  let length = WasmI32.shrS(WasmI32.fromGrain(map.size), 1n)
+  // TODO: Removing parens around Array<a> causes a parse error
+  let array = WasmI32.toGrain(allocateArray(length)): (Array<a>)
+  let setInArray = setInArray(array)
+  reduce(setInArray, 0, map)
```

Comment:
Why don't we need to free the result of reduce?

Reply:
It's a simple number, so there's no need to do anything here.